proposals/0165-async-vote-execution.md

```diff
@@ -131,12 +131,13 @@ dead later. Currently a block can be dead for the following reasons:
 
 1. Unable to load data from blockstore
 2. Invalid block (wrong number of ticks, duplicate block, bad last fec, etc)
-3. Error while set root
+3. Error while performing set root on the new root
 4. Invalid transaction
 
-For the first two, the same check can be performed computing ephemeral hash.
-We will set root on a bank only when it has full hash computed later, so the
-behavior will be the same as now.
+For the first three, the same check can be performed computing vote only
+hash. We will add a new check that the new root must be vote only replayed
+and fully replayed, this may mean the tower has more than 32 slots
```

Comment:
Can you explain this part? We avoid setting the root if the block has not been fully replayed, so the tower can contain thousands of slots?
Why can't we set the root like we do today and modify pruning to keep up to the latest fully replayed root? Essentially moving the smr to prune by to be the replayed-smr

Reply:
We need to save the slots which should be kicked out of tower but haven't been fully replayed somewhere. We can only save snapshot on a fully replayed slot, and we need to keep ancestor relationships between blocks. It could be somewhere not on the tower, are you mostly worried about the space cost if we keep it on the tower?

Reply:
we have all that in bank forks right? the tower is just a list of slot #s

Reply:
We do. I think the main question is whether we want it in vote transactions. We do want replay tip in there, but maybe not the full list of slots.